src/Microsoft.Extensions.Configuration.AzureAppConfiguration/SrvLookupClient.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DnsClient;
+using DnsClient.Protocol;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class SrvLookupClient
+    {
+        private class OriginHostCacheItem
+        {
+            public string OriginHost { get; set; }
+
+            public DateTimeOffset CacheExpires { get; set; }
+        }
+
+        private readonly ConcurrentDictionary<string, OriginHostCacheItem> _cachedOriginHosts;
```

Comment:
That looks like a cache. Why not using MemoryCache instead?

Reply:
Removed, just honor with the TTL